app/src/test/java/org/mozilla/fenix/library/bookmarks/BookmarkControllerTest.kt

```diff
@@ -36,13 +47,18 @@ import org.mozilla.fenix.components.metrics.Event
 import org.mozilla.fenix.ext.components
 
 @SuppressWarnings("TooManyFunctions", "LargeClass")
+@ExperimentalCoroutinesApi
 class BookmarkControllerTest {
 
     private lateinit var controller: BookmarkController
+    private val mainThreadSurrogate = newSingleThreadContext("UI thread")
```

Comment:
I think `MainCoroutineRule` would be sufficient here. Android Components uses it a lot to mock the main dispatcher, you can see example usage there.

Reply:
It seems like `MainCoroutineRule` isn't needed either. Tests seem to pass without it anyway